src/webgl-preview.ts

```diff
@@ -339,6 +389,39 @@ export class WebGLPreview {
     this.renderer.render(this.scene, this.camera);
   }
 
+  // create a new render method to use an animation loop to render the layers incrementally
+  async renderAnimated(layerCount = 1): Promise<void> {
+    this.renderLayerIndex = 0;
+    return this.renderFrameLoop(layerCount > 0 ? layerCount : 1);
+  }
+
+  renderFrameLoop(layerCount: number): Promise<void> {
+    return new Promise((resolve) => {
+      const loop = () => {
+        if (this.renderLayerIndex > this.layers.length - 1) {
+          resolve();
+          console.log('done rendering');
+        } else {
+          this.renderFrame(layerCount);
+          requestAnimationFrame(loop);
+        }
+      };
+      loop();
+    });
+  }
+
+  renderFrame(layerCount: number): void {
+    this.group = this.createGroup('layer' + this.renderLayerIndex);
+
+    for (let l = 0; l < layerCount && this.renderLayerIndex + l < this.layers.length; l++) {
+      this.state = this.prevState ?? State.initial;
+      this.renderLayer(this.renderLayerIndex);
+      this.prevState = { ...this.state };
+      this.renderLayerIndex++;
+    }
+    this.scene.add(this.group);
+  }
+
```

Comment:
I wonder if that should be private or not.

Reply:
Maybe also splitting it out to a separate class